fronted/src/components/PopupWithForm.js

```diff
@@ -1,12 +1,32 @@
 import React from "react";
 import './PopupWithForm.css'
+import FormValidator from "../utils/Validation";
 
 function PopupWithForm(props) {
+  const formRef = React.useRef();
+
+
+  React.useEffect(() => {
+    if (props.isValidatedForm) {
+      const settings = {
+        inputSelector: ".popup__form-input",
+        submitButtonSelector: ".popup__save-button",
+        inactiveButtonClass: "popup__save-button_disabled",
+        inputErrorClass: "popup__form-input_type_error",
+        errorClass: "popup__form-input-error_active",
+      };
+      const validatedForm = new FormValidator(settings, formRef.current);
+      validatedForm.enableValidation();
```

Comment:
this validation is a `js`-variant. But for `React` you need to make it some other way.

* `enableValidation` should be called only 1 time for each instance of `FormValidator` to enable validation. You should not do that when you open a popup (or rerender it) because you add more and more the same listeners to the same forms. This can cause a `memory leak`